frontend/ui/widget/multiinputdialog.lua

```diff
@@ -110,6 +110,13 @@ function MultiInputDialog:init()
     self:onCloseKeyboard()
     self._input_widget:onCloseWidget()
 
+    -- Initialize keyboard_visible
+    if (Device:hasKeyboard() or Device:hasScreenKB()) and G_reader_settings:isFalse("virtual_keyboard_enabled") then
+        self.keyboard_visible = false
+    else
+        self.keyboard_visible = true
+    end
```

Comment:
This should already be handled via the call to the super class's init (i.e., `InputDialog.init(self)`), whihc handles the `false` branch, couplmed with the fact that the (super) class default is `true`.

Reply:
I believe that `onCloseKeyboard` kills the tracking because, any further attempt to call `self.keyboard_visible` is always nil

Reply:
I believe that `onCloseKeyboard` kills the tracking because, any further attempt to call `self.keyboard_visible` is always nil

Reply:
disregard the previous comment. however `self.keyboard_visible` is simply `true` even when it shouldn't

```lua
    logger.dbg("MultiInputDialog:", self.keyboard_visible, "1")
    -- init title and buttons in base class
    InputDialog.init(self)
    logger.dbg("MultiInputDialog:", self.keyboard_visible, "2")
    -- Kick InputDialog's own field out of the layout, we're not using it
    table.remove(self.layout, 1)
    logger.dbg("MultiInputDialog:", self.keyboard_visible, "3")
    -- Also murder said input field *and* its VK, or we get two of them and shit gets hilariously broken real fast...
    self:onCloseKeyboard()
    logger.dbg("MultiInputDialog:", self.keyboard_visible, "4")
    self._input_widget:onCloseWidget()
    logger.dbg("MultiInputDialog:", self.keyboard_visible, "5")

    -- Initialize keyboard_visible
    -- if (Device:hasKeyboard() or Device:hasScreenKB()) and G_reader_settings:isFalse("virtual_keyboard_enabled") then
    --     self.keyboard_visible = false
    -- else
    --     self.keyboard_visible = true
    -- end
    logger.dbg("MultiInputDialog:", self.keyboard_visible, "6")
```

```
12/28/24-22:33:56 DEBUG MultiInputDialog: true 1 
12/28/24-22:33:57 DEBUG MultiInputDialog: false 2 
12/28/24-22:33:57 DEBUG MultiInputDialog: false 3 
12/28/24-22:33:57 DEBUG MultiInputDialog: true 4 
12/28/24-22:33:57 DEBUG MultiInputDialog: true 5 
12/28/24-22:33:57 DEBUG MultiInputDialog: true 6 
```

Reply:
or should we do this instead?

```lua
function InputDialog:onCloseKeyboard()
    self._input_widget:onCloseKeyboard()
    if (Device:hasKeyboard() or Device:hasScreenKB()) and G_reader_settings:isFalse("virtual_keyboard_enabled") then
        return
    end
    self.keyboard_visible = self._input_widget:isKeyboardVisible()
end
```